solidity/contracts/bridge/DepositorProxy.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.17;
+
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+
+import "./BitcoinTx.sol";
+import "./Bridge.sol";
+import "./Deposit.sol";
+import "./../vault/TBTCVault.sol";
+import "./../vault/TBTCOptimisticMinting.sol";
+import "./../token/TBTC.sol";
+
+// TODO: Make it safe for upgradeable contracts.
+// TODO: Document the contract.
+// TODO: Move to another directory?
+abstract contract DepositorProxy {
+    using BTCUtils for bytes;
+
+    /// @notice Multiplier to convert satoshi to TBTC token units.
+    uint256 public constant SATOSHI_MULTIPLIER = 10**10;
+
+    Bridge public bridge;
+    TBTCVault public tbtcVault;
+    TBTC public tbtcToken;
```

Comment:
Do we need it? It doesn't seem to be used.

Reply:
Yeah, we don't. I was initially thinking about a way to manage the TBTC balance from the abstract contract level but it seems better to push that responsibility to the child contracts. See 8c068d4f